frontend/src/metabase/querying/filters/components/DatePicker/SpecificDatePicker/DateRangePicker/DateRangePickerBody/DateRangePickerBody.tsx

```diff
@@ -92,7 +97,7 @@ export function DateRangePickerBody({
       )}
       <DatePicker
         type="range"
-        value={[startDate, hasEndDate ? endDate : null]}
+        value={inProgressDateRange ?? value}
```

Comment:
I think we should disable the "Update filter" button if range selection is in progress.
It will submit the previous valid selection, not the current in-progress one.

Reply:
@mngr maybe this is something we want to follow-up with. I'm personally not sure if that would be an improvement since I don't expect most people wouldn't face issues with the existing approach and the disabled button can only bring more confusion.